nav2_mppi_controller/src/critics/constraint_critic.cpp

```diff
@@ -50,40 +48,31 @@ void ConstraintCritic::score(CriticData & data)
   auto diff = dynamic_cast<DiffDriveMotionModel *>(data.motion_model.get());
   if (diff != nullptr) {
     if (power_ > 1u) {
-      data.costs += xt::pow(
-        xt::sum(
-          (std::move(
-            xt::maximum(data.state.vx - max_vel_, 0.0f) +
-            xt::maximum(min_vel_ - data.state.vx, 0.0f))) *
-          data.model_dt, {1}, immediate) * weight_, power_);
+      data.costs += (((((data.state.vx - max_vel_).max(0.0f) + (min_vel_ - data.state.vx).
+        max(0.0f)) * data.model_dt).rowwise().sum().eval()) * weight_).pow(power_).eval();
     } else {
-      data.costs += xt::sum(
-        (std::move(
-          xt::maximum(data.state.vx - max_vel_, 0.0f) +
-          xt::maximum(min_vel_ - data.state.vx, 0.0f))) *
-        data.model_dt, {1}, immediate) * weight_;
+      data.costs += (((((data.state.vx - max_vel_).max(0.0f) + (min_vel_ - data.state.vx).
+        max(0.0f)) * data.model_dt).rowwise().sum().eval()) * weight_).eval();
     }
     return;
   }
 
   // Omnidirectional motion model
   auto omni = dynamic_cast<OmniMotionModel *>(data.motion_model.get());
   if (omni != nullptr) {
-    auto sgn = xt::eval(xt::where(data.state.vx > 0.0f, 1.0f, -1.0f));
-    auto vel_total = sgn * xt::hypot(data.state.vx, data.state.vy);
+    auto & vx = data.state.vx;
+    unsigned int n_rows = data.state.vx.rows();
+    unsigned int n_cols = data.state.vx.cols();
+    Eigen::ArrayXXf sgn(n_rows, n_cols);
+    sgn = vx.unaryExpr([](const float x){return copysignf(1.0f, x);});
+
+    auto vel_total = (data.state.vx.square() + data.state.vy.square()).sqrt() * sgn;
```

Comment:
More an ergonomic thing, but I'd like the `sgn *` before the rest of the expression. It reads more naturally IMO